### PR TITLE
User and User Profile

### PR DESCRIPTION
**Pull request by EduardoABG**

### CODE REVIEW

File: apps/backend/src/modules/user/user.controller.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  })
+  @ApiCreatedResponse({
+    type: CreateUserResponseEntity,
+    status: HttpStatus.CREATED,
+    description: 'User successfully created.',
+  })
+  create(
+    @CognitoUser()
+    { sub: idRefAuth, email }: CognitoUserAttributes,
+    @Body() { name, accountType }: CreateUserDto
```
Comment (Bidfca) [inline apps/backend/src/modules/user/user.controller.ts:59]: The profile picture, which the user can provide during sign up, is missing

Comment (Bidfca) [reply apps/backend/src/modules/user/user.controller.ts:59]: Unless you're considering making two API calls, one for 'create user' and another for 'create profile'

File: apps/backend/src/modules/user/user.service.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        const { key, url } = await this.fileService.uploadFile(
+          profilePhoto[0],
+          'public-read'
+        );
+        pictureKeyAndUrl.push({ key, url });
+        updateUserProfileData.profilePhoto = {
+          upsert: { create: { key, url }, update: { key, url } },
+        };
+      }
+      const createResponse = await this.prismaService.userProfile.update({
```
Comment (Bidfca) [inline apps/backend/src/modules/user/user.service.ts:126]: Shouldn't it be updateResponse?

**Review by Bidfca — CHANGES_REQUESTED** [review]

File: apps/backend/src/modules/user/dto/create-profile.dto.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+      value = JSON.parse(value);
+      return plainToInstance(CreateUserProfessionalExperienceDto, value);
+    }
+    if (!Array.isArray(value)) value = [value];
+    return plainToInstance(CreateUserProfessionalExperienceDto, value);
+  })
+  @IsArray()
+  @IsObject({ each: true })
+  @ValidateNested({ each: true })
+  @Type(() => CreateUserProfessionalExperienceDto)
```
[the remark this answers is not in the document]
Comment (EduardoABG) [reply apps/backend/src/modules/user/dto/create-profile.dto.ts:110]: I forgot to remove the "@Type"

File: apps/backend/src/modules/user/user.controller.ts (hunk 1 above)
Comment (EduardoABG) [reply apps/backend/src/modules/user/user.controller.ts:59]: This will be removed the picture in the signup

**Pull request merged**